tasks/custom_task/custom_task.py

```diff
@@ -0,0 +1,81 @@
+"""
+This module defines the InvokeLogger context manager to log the invoke task information to the DD_INVOKE_LOGS_PATH.
+This will then be uploaded to Datadog's backend with a correct Log Agent configuration.
+"""
+
+from time import perf_counter
+from getpass import getuser
+from datetime import datetime
+import logging
+import traceback
+from invoke import Context
+
+DD_INVOKE_LOGS_PATH = "/tmp/dd_invoke.log"
+
+
+def log_invoke_task(name: str, module: str, task_datetime: str, duration: float, task_result: str) -> None:
+    """
+    Logs the task information to the DD_INVOKE_LOGS_PATH file.
+    This should be uploaded to Datadog's backend with a correct Log Agent configuration:
+    ```
+    logs:
+    - type: file
+        path: "/tmp/dd_invoke.log"
+        service: "dd_invoke_logs"
+        source: "invoke"
+    ```
+    """
+    logging.basicConfig(filename=DD_INVOKE_LOGS_PATH, level=logging.INFO, format='%(message)s')
+    user = getuser()
+    task_info = {
+        "name": name,
+        "module": module,
+        "datetime": task_datetime,
+        "duration": duration,
+        "user": user,
+        "result": task_result,
+    }
+    logging.info(task_info)
+
+
+class InvokeLogger:
+    """
+    Context manager to log an invoke task run information.
+    """
+
+    def __init__(self, task) -> None:
+        self.task = task
+        self.datetime = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        self.start = None
+        self.end = None
+
+    def __enter__(self):
+        self.start = perf_counter()
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self.end = perf_counter()
+        duration = round(self.end - self.start, 4)
+        name = self.task.__name__
+        module = self.task.__module__.replace("tasks.", "")
+        task_result = (
+            None if exc_type is None else "".join(traceback.format_exception(exc_type, exc_value, exc_traceback))
+        )
+        log_invoke_task(
```

Comment:
What happens if this call fails for any reason? Will it prevent the user from using any invoke tasks. Maybe we could think of something that would not impact user if the custom call fail somewhere that is not expected? WDYT?

Reply:
It seems to fail at the end of the execution of the task, so it might not prevent it from being launched but might surprise users

Reply:
That's a good point ! I'll make the `log_invoke_task` failure transparent so that it doesn't bother the current way of working.

Reply:
> so it might not prevent it from being launched
I actually think it does prevent the task to run correctly since there's code that needs to be run after the `with` statement.

Reply:
This is now fixed, all the `InvokeLogger` class logic is wrapped with a `try:...except:...` block not to change the invoke task behavior